com.unity.cinemachine/Editor/Samples/SampleDependencyImporter.cs

```diff
@@ -138,17 +155,71 @@ void ImportPackageDependencies()
+                    {
+                        var copyTo = 
+                            $"{Application.dataPath}/Samples/{packageInfo.displayName}/{packageInfo.version}/{path}";
+                        reimport = Directory.Exists(copyTo);
+                        CopyDirectory(dependencyPath, copyTo);
+                        assetsImported = true;
+                    }
+                }
+
+                return assetsImported;
+            }
+
+            static void DeleteSampleImported(string importPath, 
+                bool sharedAssetsImported, string[] sharedAssetDependencies, 
+                bool localAssetsImported, string[] localAssetDependencies)
             {
-                return EditorUtility.DisplayDialog(
-                    "Import Sample Package Dependencies",
-                    "These samples contain package dependencies that your project may not have: \n" +
-                    dependencies.Aggregate("", (current, dependency) => current + (dependency + "\n")) +
-                    "\nRemark: If your project has these dependencies, they will be updated to the latest recommended version.",
-                    "Yes, import package dependencies", "No, do not import package dependencies");
+                AssetDatabase.Refresh();
+                var relativePath = importPath.Substring(importPath.IndexOf(k_SampleName));
+                var version = relativePath.Substring(k_SampleName.Length);
+                version = version.Substring(0, version.IndexOf("/") + 1);
+
+                var instructions = new List<string>();
+                if (sharedAssetsImported)
+                    foreach (var folder in sharedAssetDependencies) 
+                        instructions.Add("Assets/" + k_SampleName + version + folder);
+                                                        
+                if (localAssetsImported)
+                    foreach (var folder in localAssetDependencies)
+                        instructions.Add("Assets/" + k_SampleName + version + folder);
+                                                        
+                instructions.Add("Assets/" + relativePath);
+
+                AssetDatabase.DeleteAssets(instructions.ToArray(), new List<string>());
+                AssetDatabase.Refresh();
             }
         }
-        
+
+        enum UserChoice
+        {
+            ImportPackageDependencies = 0,
+            ImportSampleButNotPackageDependencies = 1,
+            AbortImport = 2
+        }
+        static UserChoice PromptUserConfirmation(List<string> dependencies)
+        {
+            return (UserChoice) EditorUtility.DisplayDialogComplex(
+                "Import Sample Package Dependencies",
+                "These samples contain package dependencies that your project does not have: \n" +
+                dependencies.Aggregate("", (current, dependency) => current + (dependency + "\n")),
+                "Import samples and its dependencies", 
```

Comment:
I would suggest:

- Import samples and their dependencies
- Import samples without their dependencies
- Cancel importing the samples